spec.md

```diff
@@ -0,0 +1,1311 @@
+
+**Case CARD**
+* `A relates I @card(n..m)` postulates $n \leq k \leq m$ whenever $`a : A'(\{...\} : I^k)`$, $`A' \leq A`$, $`A' : \mathbf{Rel}(I)`$, and $k$ is _maximal_ (for fixed $a : A$).
+  * **defaults** to `@card(1..1)` if omitted ("one")
+* `A plays B:I @card(n..m)` postulates $n \leq |B(a:I)| \leq m$ for all $a : A$
+  * **defaults** to `@card(0..)` if omitted ("many")
+* `A owns B @card(n...m)` postulates $n \leq |B(a:I)| \leq m$ for all $a : A$
+  * **defaults** to `@card(0..1)` if omitted ("one or null")
+
+***System property***:
+
+1. For inherited interfaces, we cannot redeclare cardinality (this is actually a consequence of "Implicit inheritance" above). 
+2. When we have direct subinterfaces $I_i <_! J$, for $i = 1,...,n$, and each $I_i$ has `card(`$`n_i,m_i`$`)` while J has $card(n,m)$ then we must have $`n \leq \sum_i n_i \leq \sum_i m_i \leq m`$.
+  
+_Remark 1: Upper bounds can be omitted, writing `@card(2..)`, to allow for arbitrary large cardinalities_
+
+_Remark 2: For cardinality, and for most other constraints, we should reject redundant conditions, such as `A owns B card(0..3);` when `A sub A'` and `A' owns B card(1..2);`_
+
+**Case CARD_LIST**
+* `A relates I[] @card(n..m)` postulates $n \leq \mathrm{len}(l) \leq m$ whenever $a : A'(l : [I])$, $A' \leq A$, $A' : \mathbf{Rel}([I])$, and $k$ is _maximal_ (for fixed $a : A$).
+  * **defaults** to `@card(0..)` if omitted ("many")
+* `A owns B[] @card(n...m)` postulates $n \leq \mathrm{len}(l) \leq m$ whenever $`l : [B](a:O_B)`$ for $`a : A`$
+  * **defaults** to `@card(0..)` if omitted ("many")
+
+**Case PLAYS_AS**
+* `A plays B as C` postulates $`|C(x:O_C)| - |B(x:O_B)| = 0`$ for all $`x:A`$, ***requiring*** that $B \lneq C$, $A < D$, $`D <_! C`$. **Invalidated** when $A <_! C'$ for $B \lneq C' \leq C$.
+
+**Case OWNS_AS**
+* `A owns B as C` postulates $`|C(x:O_C)| - |B(x:O_B)| = 0`$ for all $`x:A`$, ***requiring*** that $B \lneq C$, $A < D$, $`D <_! O_C`$. **Invalidated** when $A <_! O_{C'}$ for $B \lneq C' \leq C$.
+
+_Comment: both preceding cases are kinda complicated/unnatural ... as reflected by the math._
+
+**Case UNIQUE**
+* `A owns B @unique` postulates that if $`b : B(a:O_B)`$ for some $a : A$ then this $a$ is unique (for fixed $b$).
+
+**Case KEY**
+* `A owns B @key` postulates that if $`b : B(a:O_B)`$ for some $a : A$ then this $a$ is unique, and also $|B(a:O_B) = 1$.
+
+**Case SUBKEY**
+* `A owns B1 @subkey(<LABEL>); A owns B2 @subkey(<LABEL>)` postulates that if $b : B_1(a:O_{B_1}) \times B_2(a:O_{B_2})`$ for some $a : A$ then this $a$ is unique, and also $|B_1(a:O_{B_1}) \times B_2(a:O_{B_2})| = 1$. **Generalizes** to $n$ subkeys.
+
+**Case ABSTRACT**
+* `(type) A @abstract` postulates $`a :_! A(...)`$ to be impossible
+* `B relates I @abstract` postulates $`A <_! I`$ to be impossible for $A : \mathbf{Obj}$
+* `B relates I[] @abstract` postulates $`A <_! I`$ to be impossible for $A : \mathbf{Obj}$
+* `A plays B:I @abstract` postulates that
```

Comment:
Unclear why there are two cases here. Does this mean that in `plays` statements we would not distinguish between list and set roles? This doesn't feel consistent with other usage if the case. Perhaps needs some thought.